{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2021-05-29T12:04:12.421Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-05-29T12:04:09.091235Z",
     "start_time": "2021-05-29T12:04:08.206595Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>incident_id</th>\n",
       "      <th>date</th>\n",
       "      <th>state</th>\n",
       "      <th>city_or_county</th>\n",
       "      <th>n_killed</th>\n",
       "      <th>n_injured</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>483817</td>\n",
       "      <td>2013-03-31</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Auburn</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>47.3149</td>\n",
       "      <td>-122.224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>484333</td>\n",
       "      <td>2013-04-21</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Federal Way</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>47.3030</td>\n",
       "      <td>-122.310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>491281</td>\n",
       "      <td>2013-07-29</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Granger</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>46.3421</td>\n",
       "      <td>-120.187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193</th>\n",
       "      <td>480071</td>\n",
       "      <td>2013-09-15</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Yakima</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>46.5854</td>\n",
       "      <td>-120.538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>323</th>\n",
       "      <td>92345</td>\n",
       "      <td>2014-01-01</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Georgetown</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>47.5498</td>\n",
       "      <td>-122.318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239176</th>\n",
       "      <td>1079791</td>\n",
       "      <td>2018-03-28</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Seattle</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47.6138</td>\n",
       "      <td>-122.332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239399</th>\n",
       "      <td>1080903</td>\n",
       "      <td>2018-03-29</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Pasco</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239401</th>\n",
       "      <td>1081610</td>\n",
       "      <td>2018-03-29</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Walla Walla</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>46.0737</td>\n",
       "      <td>-118.312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239402</th>\n",
       "      <td>1081612</td>\n",
       "      <td>2018-03-29</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Yakima</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>46.6173</td>\n",
       "      <td>-120.512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239671</th>\n",
       "      <td>1082061</td>\n",
       "      <td>2018-03-31</td>\n",
       "      <td>Washington</td>\n",
       "      <td>Spokane (Spokane Valley)</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>47.6638</td>\n",
       "      <td>-117.235</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3434 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        incident_id        date       state            city_or_county  \\\n",
       "45           483817  2013-03-31  Washington                    Auburn   \n",
       "54           484333  2013-04-21  Washington               Federal Way   \n",
       "153          491281  2013-07-29  Washington                   Granger   \n",
       "193          480071  2013-09-15  Washington                    Yakima   \n",
       "323           92345  2014-01-01  Washington                Georgetown   \n",
       "...             ...         ...         ...                       ...   \n",
       "239176      1079791  2018-03-28  Washington                   Seattle   \n",
       "239399      1080903  2018-03-29  Washington                     Pasco   \n",
       "239401      1081610  2018-03-29  Washington               Walla Walla   \n",
       "239402      1081612  2018-03-29  Washington                    Yakima   \n",
       "239671      1082061  2018-03-31  Washington  Spokane (Spokane Valley)   \n",
       "\n",
       "        n_killed  n_injured  latitude  longitude  \n",
       "45             3          2   47.3149   -122.224  \n",
       "54             5          0   47.3030   -122.310  \n",
       "153            0          4   46.3421   -120.187  \n",
       "193            0          6   46.5854   -120.538  \n",
       "323            0          1   47.5498   -122.318  \n",
       "...          ...        ...       ...        ...  \n",
       "239176         0          0   47.6138   -122.332  \n",
       "239399         0          1       NaN        NaN  \n",
       "239401         0          0   46.0737   -118.312  \n",
       "239402         0          1   46.6173   -120.512  \n",
       "239671         0          0   47.6638   -117.235  \n",
       "\n",
       "[3434 rows x 8 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('incidents.csv')\n",
    "wa=df[df['state'] == 'Washington']\n",
    "wa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2021-05-29T12:04:12.863973Z",
     "start_time": "2021-05-29T12:04:09.100974Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Pandarallel will run on 32 workers.\n",
      "INFO: Pandarallel will use Memory file system to transfer data between the main process and workers.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[nltk_data] Downloading package vader_lexicon to\n",
      "[nltk_data]     /home/yeraf/nltk_data...\n",
      "[nltk_data]   Package vader_lexicon is already up-to-date!\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import datetime\n",
    "from pandarallel import pandarallel\n",
    "from nltk.sentiment.vader import SentimentIntensityAnalyzer\n",
    "pandarallel.initialize(nb_workers=32)\n",
    "import twint\n",
    "import time\n",
    "import random\n",
    "import nltk\n",
    "import nest_asyncio\n",
    "nest_asyncio.apply()\n",
    "__import__('IPython').embed()\n",
    "\n",
    "nltk.download('vader_lexicon')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "ExecuteTime": {
     "start_time": "2021-05-29T12:04:06.929Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "error:  This event loop is already running\n",
      "error:  This event loop is already running\n",
      "error:  This event loop is already running\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Process ForkPoolWorker-33:\n",
      "Process ForkPoolWorker-32:\n",
      "Process ForkPoolWorker-30:\n",
      "Process ForkPoolWorker-28:\n",
      "Process ForkPoolWorker-27:\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-29:\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-31:\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-26:\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Process ForkPoolWorker-25:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Process ForkPoolWorker-23:\n",
      "Process ForkPoolWorker-16:\n",
      "Process ForkPoolWorker-12:\n",
      "Process ForkPoolWorker-24:\n",
      "Process ForkPoolWorker-20:\n",
      "Process ForkPoolWorker-11:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Traceback (most recent call last):\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Process ForkPoolWorker-15:\n",
      "RuntimeError: This event loop is already running\n",
      "Traceback (most recent call last):\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Process ForkPoolWorker-21:\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-22:\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-18:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Process ForkPoolWorker-6:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "RuntimeError: This event loop is already running\n",
      "Process ForkPoolWorker-2:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "Process ForkPoolWorker-7:\n",
      "Process ForkPoolWorker-10:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Process ForkPoolWorker-14:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Process ForkPoolWorker-13:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Process ForkPoolWorker-9:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Process ForkPoolWorker-5:\n",
      "Process ForkPoolWorker-19:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Process ForkPoolWorker-3:\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "RuntimeError: This event loop is already running\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "Process ForkPoolWorker-8:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "Traceback (most recent call last):\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Process ForkPoolWorker-17:\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Process ForkPoolWorker-4:\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "Traceback (most recent call last):\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 13, in get_tweet_loc\n",
      "    twint.run.Search(c)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 410, in Search\n",
      "    run(config, callback)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "RuntimeError: This event loop is already running\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "RuntimeError: This event loop is already running\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/twint/run.py\", line 329, in run\n",
      "    get_event_loop().run_until_complete(Twint(config).main(callback))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "RuntimeError: This event loop is already running\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "Traceback (most recent call last):\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "Traceback (most recent call last):\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 475, in run_until_complete\n",
      "    self.run_forever()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "Traceback (most recent call last):\n",
      "Traceback (most recent call last):\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "Traceback (most recent call last):\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/asyncio/base_events.py\", line 429, in run_forever\n",
      "    raise RuntimeError('This event loop is already running')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "RuntimeError: This event loop is already running\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-81c4591b7121>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     60\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 62\u001b[0;31m \u001b[0mwa_polar\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mwa\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparallel_apply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mget_loca_polar\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     63\u001b[0m \u001b[0mwa_polar\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mto_csv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'wa_polar_1_14d.csv'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\u001b[0m in \u001b[0;36mclosure\u001b[0;34m(data, func, *args, **kwargs)\u001b[0m\n\u001b[1;32m    460\u001b[0m                 \u001b[0minput_files\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    461\u001b[0m                 \u001b[0moutput_files\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 462\u001b[0;31m                 \u001b[0mmap_result\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    463\u001b[0m             )\n\u001b[1;32m    464\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\u001b[0m in \u001b[0;36mget_workers_result\u001b[0;34m(use_memory_fs, nb_workers, show_progress_bar, nb_columns, queue, chunk_lengths, input_files, output_files, map_result)\u001b[0m\n\u001b[1;32m    364\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    365\u001b[0m     \u001b[0;32mwhile\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfinished_workers\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 366\u001b[0;31m         \u001b[0mmessage_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmessage\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mqueue\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    367\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    368\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmessage_type\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mINPUT_FILE_READ\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<string>\u001b[0m in \u001b[0;36mget\u001b[0;34m(self, *args, **kwds)\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/managers.py\u001b[0m in \u001b[0;36m_callmethod\u001b[0;34m(self, methodname, args, kwds)\u001b[0m\n\u001b[1;32m    755\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    756\u001b[0m         \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethodname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 757\u001b[0;31m         \u001b[0mkind\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconn\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    758\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    759\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mkind\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'#RETURN'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/connection.py\u001b[0m in \u001b[0;36mrecv\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    248\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_closed\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    249\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_check_readable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 250\u001b[0;31m         \u001b[0mbuf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_recv_bytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    251\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0m_ForkingPickler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbuf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetbuffer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    252\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/connection.py\u001b[0m in \u001b[0;36m_recv_bytes\u001b[0;34m(self, maxsize)\u001b[0m\n\u001b[1;32m    405\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    406\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_recv_bytes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmaxsize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 407\u001b[0;31m         \u001b[0mbuf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_recv\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    408\u001b[0m         \u001b[0msize\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mstruct\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munpack\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"!i\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbuf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgetvalue\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    409\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmaxsize\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0msize\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0mmaxsize\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/connection.py\u001b[0m in \u001b[0;36m_recv\u001b[0;34m(self, size, read)\u001b[0m\n\u001b[1;32m    377\u001b[0m         \u001b[0mremaining\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msize\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    378\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mremaining\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 379\u001b[0;31m             \u001b[0mchunk\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mremaining\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    380\u001b[0m             \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mchunk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    381\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "KeyboardInterrupt\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 119, in worker\n",
      "    result = (True, func(*args, **kwds))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "KeyboardInterrupt\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/lib/python3.6/multiprocessing/pool.py\", line 44, in mapstar\n",
      "    return list(map(*args))\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/data_types/dataframe.py\", line 31, in worker\n",
      "    return df.apply(func, *args, **kwargs)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 64, in global_worker\n",
      "    return _func(x)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/frame.py\", line 7552, in apply\n",
      "    return op.get_result()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "KeyboardInterrupt\n",
      "KeyboardInterrupt\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandarallel/pandarallel.py\", line 116, in wrapper\n",
      "    **kwargs\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 185, in get_result\n",
      "    return self.apply_standard()\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "KeyboardInterrupt\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "KeyboardInterrupt\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 276, in apply_standard\n",
      "    results, res_index = self.apply_series_generator()\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 33, in get_tweet_loc\n",
      "    time.sleep(random.random() * 300)\n",
      "  File \"/home/yeraf/.pyenv/versions/3.6.10/envs/us-gun/lib/python3.6/site-packages/pandas/core/apply.py\", line 305, in apply_series_generator\n",
      "    results[i] = self.f(v)\n",
      "  File \"<ipython-input-4-81c4591b7121>\", line 46, in get_loca_polar\n",
      "    \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n"
     ]
    }
   ],
   "source": [
    "radius = 2\n",
    "over = 14\n",
    "def get_tweet_loc(params):\n",
    "    for _ in range(10):\n",
    "        try:\n",
    "            c = twint.Config()\n",
    "            c.Search = params['keyword']\n",
    "            c.Limit = 100\n",
    "            c.Geo = f\"{str(params['lat'])},{str(params['lng'])},{str(params['distance'])}{params['unit']}\"\n",
    "            c.Pandas = True\n",
    "            c.Since = params['since']\n",
    "            c.Until = params['until']\n",
    "            twint.run.Search(c)\n",
    "            sid = SentimentIntensityAnalyzer()\n",
    "            stat = {\"pos\": 0, \"neg\": 0, \"neu\": 0, \"comp\": 0, \"scrape\": True}\n",
    "            print(twint.storage.panda.Tweets_df.shape)\n",
    "            if twint.storage.panda.Tweets_df.shape[0] > 0:\n",
    "                tweets = twint.storage.panda.Tweets_df['tweet']\n",
    "                for tweet in tweets:\n",
    "                    senvalue = sid.polarity_scores(tweet)\n",
    "                    stat['pos'] += senvalue[\"pos\"]\n",
    "                    stat['neg'] += senvalue[\"neg\"]\n",
    "                    stat['neu'] += senvalue[\"neu\"]\n",
    "                    stat['comp'] += senvalue[\"compound\"]\n",
    "                stat['neg'] /= tweets.shape[0]\n",
    "                stat['pos'] /= tweets.shape[0]\n",
    "                stat['neu'] /= tweets.shape[0]\n",
    "                stat['comp'] /= tweets.shape[0] \n",
    "            return stat\n",
    "        except Exception as e:\n",
    "            time.sleep(random.random() * 300)\n",
    "            print('error: ', e)\n",
    "            continue\n",
    "    return {\"pos\": 0, \"neg\": 0, \"neu\": 0, \"comp\": 0, \"scrape\": True}\n",
    "\n",
    "def get_loca_polar(r):\n",
    "    response = get_tweet_loc({\n",
    "            \"keyword\": \"gun\",\n",
    "            \"lat\": r['latitude'],\n",
    "            \"lng\": r['longitude'],\n",
    "            \"distance\": radius,\n",
    "            \"unit\": \"km\",\n",
    "            \"since\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') - datetime.timedelta(days=over)).strftime('%Y-%m-%d'),\n",
    "            \"until\": (datetime.datetime.strptime(r['date'], '%Y-%m-%d') + datetime.timedelta(days=over)).strftime('%Y-%m-%d')\n",
    "        })\n",
    "    print(response)\n",
    "    if response:\n",
    "        r['pos'] = response['pos']\n",
    "        r['neg'] = response['neg']\n",
    "        r['neu'] = response['neu']\n",
    "        r['comp'] = response['comp']\n",
    "        r['scrape'] = response['scrape']\n",
    "    else:\n",
    "        r['pos'] = 0\n",
    "        r['neg'] = 0\n",
    "        r['neu'] = 0\n",
    "        r['comp'] = 0\n",
    "        return r\n",
    "\n",
    "wa_polar = wa.parallel_apply(get_loca_polar, axis=1)\n",
    "wa_polar.to_csv('wa_polar_1_14d.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
